### PR TITLE
CRM-16189, added function to validate financial type

### PR DESCRIPTION
**Pull request by pradpnayak**
---
- CRM-16189: Improve support for Accrual Method bookkeeping
  https://issues.civicrm.org/jira/browse/CRM-16189

### CODE REVIEW

File: CRM/Financial/BAO/FinancialAccount.php [hunk 1]
```diff
@@ -246,4 +246,48 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
...
+   * check if financial type has Deferred Revenue Account is relationship
+   * with Financial Account
+   *
+   * @param $financialTypeId Integer
+   *
+   * @return string
+   *
+   */
+  public static function validateFinancialType($financialTypeId, $entityID = NULL, $entity = NULL) {
+    if (!$financialTypeId) {
```
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: this check doesn't look necessary

File: CRM/Financial/BAO/FinancialAccount.php [hunk 2]
```diff
@@ -246,4 +246,48 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * check if financial type has Deferred Revenue Account is relationship
```
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: The full standards include a capital letter & a full stop - which is preferable

**Comment by eileenmcnaughton** [comment]
As with the other PR - these are nice testable functions - let's add the tests right from the start

**Comment by pradpnayak** [comment]
Added unit test

File: CRM/Financial/BAO/FinancialAccount.php (hunk 1 above)
Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: It does, when updating Event, MembershipType etc we might not send financial_type_id in $params via api or BAO function. The function should return false in such situation.

**Comment by pradpnayak** [comment]
depends on https://github.com/civicrm/civicrm-core/pull/8562

File: CRM/Financial/BAO/FinancialAccount.php (hunk 1 above)
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: Hmm - it still feels like why would you call this function if you didn't have a financial type to validate. It just feels like it's encouraging bad coding to do this.

Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: (NB that's an opinion - not a blocker)

**Comment by pradpnayak** [comment]
@eileenmcnaughton / @colemanw can you please review this PR?

File: tests/phpunit/CRM/Financial/BAO/FinancialAccountTest.php [hunk 3]
```diff
@@ -267,6 +267,7 @@ public function testValidateFinancialAccount() {
   }
 
   /**
+<<<<<<< HEAD
```
Comment (seamuslee001) [inline tests/phpunit/CRM/Financial/BAO/FinancialAccountTest.php]: @pradpnayak This needs to be removed

**Comment by colemanw** [comment]
@pradpnayak I don't understand the return signature of this function. The docblock says it returns a string but it is returning a boolean that is never not false, except for when it returns a specific message. It seems strange for a utility function like this one to return a text message. I would expect the calling function to generate this message instead. I recommend changing the text message returned by this funciton to be `TRUE` and note in the docblock that it will always return a bool.

**Comment by pradpnayak** [comment]
@colemanw i added the changes, Could you please review now?

**Comment by colemanw** [comment]
Looks good.

**Pull request merged**